src/vm/amd64/profiler.cpp

```diff
@@ -41,6 +41,11 @@ typedef struct _PROFILE_PLATFORM_SPECIFIC_DATA
     UINT64      flt5;
     UINT64      flt6;
     UINT64      flt7;
+    UINT64      rdi;
```

Comment:
Hmm, then there is a deeper issue I haven't realized before. Since the ProfileEnterNaked is called with the first two argument registers already containing different values than the parameters of the profiled function, we need something out of this helper to save the real arguments somewhere before the ProfileEnterNaked is called.
@noahfalk, @CarolEidt how do we solve that for ARM that doesn't home argument registers even on Windows unless I am mistaken?